napari/_qt/layer_controls/qt_image_controls_base.py

```diff
@@ -158,6 +158,16 @@ def __init__(self, layer: Image) -> None:
             # widgets.
             self.layout().addRow(self.button_grid)
 
+    def _on_make_colormap(self):
+        """Make new colormap."""
+        from napari._qt.utils import get_color
+        from napari.utils.colormaps.colormap_utils import napari_colormap
+
+        color = get_color(self, as_hex=True)
+        if color:
+            colormap = napari_colormap(color, name=color)
```

Comment:
```suggestion
        from napari.utils.colormaps.colormap_utils import ensure_colormap

        color = get_color(self, as_hex=True)
        if color:
            colormap = ensure_colormap(color)
```

instead of the new `napari_colormap` can use the existing `ensure_colormap`. For HEX codes this will first check if that's already a colormap.
Obviously it won't catch random colors, but those that are registered will be re-used.

Reply:
Thanks! A nice catch and removes some redundant code.
Changed in a PR.